tests/components/evohome/test_storage.py

```diff
@@ -0,0 +1,163 @@
+
+        MockEvohomeClient.assert_called_once()
+        assert MockEvohomeClient.call_args.args[0] == test_config[CONF_USERNAME]
+        assert MockEvohomeClient.call_args.args[1] == test_config[CONF_PASSWORD]
+
+        assert isinstance(MockEvohomeClient.call_args.kwargs["session"], ClientSession)
+        assert mock_client and mock_client.account_info is not None
+
+        return MockEvohomeClient
+
+
+@pytest.mark.parametrize("idx", TEST_DATA_NULL)
+async def test_auth_tokens_null(
+    hass: HomeAssistant,
+    hass_storage: dict[str, Any],
+    idx: str,
+) -> None:
+    """Test loading/saving authentication tokens when no cached tokens."""
+
+    hass_storage[DOMAIN] = DOMAIN_STORAGE_BASE | {"data": TEST_DATA_NULL[idx]}
+
+    MockClient = await setup_evohome(hass, TEST_CONFIG)
+
+    # Confirm the client was instantiated with the correct kwargs...
+    assert "refresh_token" not in MockClient.call_args.kwargs
+    assert "access_token" not in MockClient.call_args.kwargs
+    assert "access_token_expires" not in MockClient.call_args.kwarg
+
+    # Confirm the expected tokens were cached to storage...
+    assert hass_storage[DOMAIN]["data"][CONF_USERNAME] == USERNAME_SAME
+    assert hass_storage[DOMAIN]["data"]["refresh_token"] == f"new_{REFRESH_TOKEN}"
+    assert hass_storage[DOMAIN]["data"]["access_token"] == f"new_{ACCESS_TOKEN}"
+    assert hass_storage[DOMAIN]["data"]["access_token_expires"] > str(dt_util.now())
+
+
+@pytest.mark.parametrize("idx", TEST_DATA)
+async def test_auth_tokens_same(
+    hass: HomeAssistant, hass_storage: dict[str, Any], idx: str
+) -> None:
+    """Test loading/saving authentication tokens when same username."""
+
+    hass_storage[DOMAIN] = DOMAIN_STORAGE_BASE | {"data": TEST_DATA[idx]}
+
+    MockClient = await setup_evohome(hass, TEST_CONFIG)
+
+    # Confirm the client was instantiated with the correct kwargs...
+    assert MockClient.call_args.kwargs["refresh_token"] == REFRESH_TOKEN
+    assert MockClient.call_args.kwargs["access_token"] == ACCESS_TOKEN
+    assert MockClient.call_args.kwargs["access_token_expires"] == ACCESS_TOKEN_EXP_DTM
+
+    # Confirm the expected tokens were cached to storage...
+    assert hass_storage[DOMAIN]["data"][CONF_USERNAME] == USERNAME_SAME
+    assert hass_storage[DOMAIN]["data"]["refresh_token"] == REFRESH_TOKEN
+    assert hass_storage[DOMAIN]["data"]["access_token"] == ACCESS_TOKEN
+    assert hass_storage[DOMAIN]["data"]["access_token_expires"] == ACCESS_TOKEN_EXP_STR
+
+
+@pytest.mark.parametrize("idx", TEST_DATA)
+async def test_auth_tokens_diff(
+    hass: HomeAssistant, hass_storage: dict[str, Any], idx: str
+) -> None:
+    """Test loading/saving authentication tokens when different username."""
+
+    hass_storage[DOMAIN] = DOMAIN_STORAGE_BASE | {"data": TEST_DATA[idx]}
+
+    MockClient = await setup_evohome(hass, TEST_CONFIG | {CONF_USERNAME: USERNAME_DIFF})
+
+    # Confirm the client was instantiated with the correct kwargs...
```

Comment:
Maybe clarify in the comment that we don't expect to pass the stored tokens of user 1 to the client when logging in with user 2.

That was my interpretation of what is happening here.